src/main/java/core/basesyntax/strategy/handler/BalanceOperationHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperationHandler implements OperationHandler {
+    private final FruitDao fruitDao;
+
+    public BalanceOperationHandler(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public void process(FruitTransaction transaction) {
+        fruitDao.getStorage().put(transaction.getFruit(), transaction.getQuantity());
```

Comment:
Direct access to the storage map is a violation of encapsulation. Use the appropriate 'FruitDao' methods to interact with the storage.